### PR TITLE
Update SetIkTarget

### PR DESCRIPTION
**Pull request by matehun00**
Add a lot of info to the docs of SetIkTarget
If anyone wants to add js and c# examples, feel free to comment it down below and I'll add it
I will not add more commits to this PR (unless something needs fixing)

Most of the credit goes to @kypos on discord

### CODE REVIEW

**Comment by ahcenezdh** [comment]
For the js example it's
```javascript
SetIkTarget(PlayerPedId(), 4, 0, 0, -1849.0, -1231.0, 13.0, 0, 0, 10000);
```

and c# it's
```csharp
using static CitizenFX.Core.Native.API;
// ...

SetIkTarget(PlayerPedId(), 4, 0, 0, -1849.0, -1231.0, 13.0, 0, 0, 10000);
```

**Comment by AvarianKnight** [comment]
> For the js example it's
> 
> ```js
> SetIkTarget(PlayerPedId(), 4, 0, 0, -1849.0, -1231.0, 13.0, 0, 0, 10000);
> ```
> 
> and c# it's
> 
> ```cs
> using static CitizenFX.Core.Native.API;
> // ...
> 
> SetIkTarget(PlayerPedId(), 4, 0, 0, -1849.0, -1231.0, 13.0, 0, 0, 10000);
> ```

C# example needs to annotate its float values `-1849.0f`

File: PED/SetIkTarget.md [hunk 1]
```diff
@@ -5,18 +5,59 @@ ns: PED
 
 ```c
 // 0xC32779C16FCEECD9 0x6FE5218C
-void SET_IK_TARGET(Ped ped, int ikIndex, Entity entityLookAt, int boneLookAt, float offsetX, float offsetY, float offsetZ, Any p7, int blendInDuration, int blendOutDuration);
+void SET_IK_TARGET(Ped ped, int ikIndex, Entity entityLookAt, int boneLookAt, float offsetX, float offsetY, float offsetZ, cs_type(Any) int ikTargetFlags, int blendInDuration, int blendOutDuration);
```
Comment (AvarianKnight) [inline PED/SetIkTarget.md:8]: `cs_type(Any)` isn't needed if you're using `int`

**Review by AvarianKnight — CHANGES_REQUESTED** [review]

**Review by AvarianKnight — APPROVED** [review]

File: PED/SetIkTarget.md [hunk 2]
```diff
@@ -5,18 +5,68 @@ ns: PED
...
+	ITF_ARM_TARGET_WRT_HANDBONE	= 1, // arm target relative to the handbone
+	ITF_ARM_TARGET_WRT_POINTHELPER = 2, // arm target relative to the pointhelper
+	ITF_ARM_TARGET_WRT_IKHELPER	= 4, // arm target relative to the ikhelper
+	ITF_IK_TAG_MODE_NORMAL = 8, // use animation tags directly
+	ITF_IK_TAG_MODE_ALLOW	= 16, // use animation tags in ALLOW mode
+	ITF_IK_TAG_MODE_BLOCK	= 32, // use animation tags in BLOCK mode
+	ITF_ARM_USE_ORIENTATION	= 64 // solve for orientation in addition to position
+};
+
+```
```
Comment (PsychoShock) [inline PED/SetIkTarget.md:55]: "New" natives docs try to get a description of the actual native.

Something like: "Points the right arm to the world coordinates. Check enums for more details." then provide the enums.

File: PED/SetIkTarget.md [hunk 3]
```diff
@@ -5,18 +5,68 @@ ns: PED
...
-* **p7**: 
+* **ped**: The ped handle.
+* **ikIndex**: See ``IK_PART``.
+* **entityLookAt**: Set to 0 for no entity, offsets will be world coordinates instead.
+* **boneLookAt**: Set to -1 for no target bone.
+* **offsetX**: X-axis offset calculated from the position of ``entityLookAt``.
+* **offsetY**: Y-axis offset calculated from the position of ``entityLookAt``.
+* **offsetZ**: Z-axis offset calculated from the position of ``entityLookAt``.
+* **ikTargetFlags**: See ``IK_TARGET_FLAGS``.
 * **blendInDuration**:
```
Comment (PsychoShock) [inline PED/SetIkTarget.md:22]: No description for this one? :(

File: PED/SetIkTarget.md [hunk 4]
```diff
@@ -5,18 +5,68 @@ ns: PED
...
-* **offsetZ**: 
-* **p7**: 
+* **ped**: The ped handle.
+* **ikIndex**: See ``IK_PART``.
+* **entityLookAt**: Set to 0 for no entity, offsets will be world coordinates instead.
+* **boneLookAt**: Set to -1 for no target bone.
+* **offsetX**: X-axis offset calculated from the position of ``entityLookAt``.
+* **offsetY**: Y-axis offset calculated from the position of ``entityLookAt``.
+* **offsetZ**: Z-axis offset calculated from the position of ``entityLookAt``.
+* **ikTargetFlags**: See ``IK_TARGET_FLAGS``.
```
Comment (PsychoShock) [inline PED/SetIkTarget.md:21]: Most of the docs use only 1 backticks `` ` ``, it's not include in signature method but for better standart.

File: PED/SetIkTarget.md [hunk 5]
```diff
@@ -5,18 +5,68 @@ ns: PED
...
+```js
+// Points the right arm to the world coordinates -1849.0, -1231.0, 13.0 for 10 seconds -- Which is the end of the Del Perro Pier
+SetIkTarget(PlayerPedId(), 4, 0, 0, -1849.0, -1231.0, 13.0, 0, 0, 10000);
+```
+```cs
+using static CitizenFX.Core.Native.API;
+// ...
+
+// Points the right arm to the world coordinates -1849.0, -1231.0, 13.0 for 10 seconds -- Which is the end of the Del Perro Pier
+SetIkTarget(PlayerPedId(), 4, 0, 0, -1849.0f, -1231.0f, 13.0f, 0, 0, 10000);
```
Comment (PsychoShock) [inline PED/SetIkTarget.md:71]: Following ther comments with description, you will be able to remove the "Points the right arm to the world coordinates" part and just set "Points to -1849.0, -1231.0, 13.0 for 10 seconds"

Also the "Which is the end of the Del Perro Pier" part can be removed imo. Will just confuse users who doesnt know where / what is Del Perro.

**Review by PsychoShock — CHANGES_REQUESTED** [review]
Hey,

Nice work, left some comments imo that can be changed.

File: PED/SetIkTarget.md (hunk 3 above)
Comment (matehun00) [reply PED/SetIkTarget.md:22]: I could not find out the thing that it does, in my testing it seemed to have no effect on anything

Comment (AvarianKnight) [reply PED/SetIkTarget.md:22]: Are they even named right these look like they would be similar to other animation flags (aka [TaskPlayAnim](https://docs.fivem.net/natives/?_0xEA47FE3719165B94))

These are probably meant to be `blendInSpeed` and `blendOutSpeed`

Comment (PsychoShock) [reply PED/SetIkTarget.md:22]: Well the old param is still good, it blend in time.

And `blendOutDuration` will set the time to blend out.

I'll do some tests to confirm.

Comment (PsychoShock) [reply PED/SetIkTarget.md:22]: So after some tests and research, this is what I found:

`-1` seems to be the default values, based on [decompiled files](https://github.com/root-cause/v-decompiled-scripts/blob/afd9d9e3235fe6411c9d2c5ad7109a9ab279d042/re_duel.c#L5329).

There's other cases where R* use to set a specific time, [like here](https://github.com/root-cause/v-decompiled-scripts/blob/afd9d9e3235fe6411c9d2c5ad7109a9ab279d042/country_race_controller.c#L811).

When testing, I see that it works perfectly when setting specific time to blend out and `-1` to blend in
image

But you can see here, when testing `SetIkTarget(PlayerPedId(), 4, 0, 0, -1849.0, -1231.0, 13.0, 0, -1, -1)`

https://github.com/user-attachments/assets/04bc6efa-ba57-4ee9-9f15-6257f2fec570

So I guess it's only to be used with a specific blend out in `ms`.

File: PED/SetIkTarget.md [hunk 6]
```diff
@@ -5,18 +5,68 @@ ns: PED
...
+* **blendOutDuration**: Sets how long the animation lasts in ms.
+
+
+```
+NativeDB Introduced: v323
+```
+
+```c
+enum IK_PART {
+  K_PART_INVALID = 0,
```
Comment (AvarianKnight) [inline PED/SetIkTarget.md:32]: Didn't notice this when i reviewed earlier
```suggestion
  IK_PART_INVALID = 0,
```

**Comment by AvarianKnight** [comment]
Also not sure on the preferred way to do enums is, most use `enum eCamelCase` and its enum variables are `UpperCamelCase` (though this varies from native to native, making it hard to tell what should be done)

File: PED/SetIkTarget.md (hunk 3 above)
Comment (matehun00) [reply PED/SetIkTarget.md:22]: so what should I write for it?

File: PED/SetIkTarget.md (hunk 2 above)
Comment (matehun00) [reply PED/SetIkTarget.md]: I don't think understand correctly, where should I put something like "Points the right arm to the world coordinates. Check enums for more details." to?

**Review by PsychoShock — CHANGES_REQUESTED** [review]
---
ns: PED
---
## SET_IK_TARGET

```c
// 0xC32779C16FCEECD9 0x6FE5218C
void SET_IK_TARGET(Ped ped, int ikIndex, Entity entityLookAt, int boneLookAt, float offsetX, float offsetY, float offsetZ, int ikTargetFlags, int blendInDuration, int blendOutDuration);
```

Sets the IK target for a given IK part belonging to the ped.

**Please note:** The IK target will only be valid for one update, so it needs to be set for as long as it is needed to avoid IK targets not being cleared and getting stuck enabled.

```c
enum eIkPart {
  IK_PART_INVALID = 0,
  IK_PART_HEAD = 1, // head
  IK_PART_SPINE = 2, // spine
  IK_PART_ARM_LEFT = 3, // Left Arm
  IK_PART_ARM_RIGHT = 4, // Right Arm
  IK_PART_LEG_LEFT = 5, // Left Leg
  IK_PART_LEG_RIGHT = 6 // Right Leg
};

```

```c
enum eIkTargetFlags {
	ITF_DEFAULT = 0,
	ITF_ARM_TARGET_WRT_HANDBONE	= 1, // arm target relative to the handbone
	ITF_ARM_TARGET_WRT_POINTHELPER = 2, // arm target relative to the pointhelper
	ITF_ARM_TARGET_WRT_IKHELPER	= 4, // arm target relative to the ikhelper
	ITF_IK_TAG_MODE_NORMAL = 8, // use animation tags directly
	ITF_IK_TAG_MODE_ALLOW	= 16, // use animation tags in ALLOW mode
	ITF_IK_TAG_MODE_BLOCK	= 32, // use animation tags in BLOCK mode
	ITF_ARM_USE_ORIENTATION	= 64 // solve for orientation in addition to position
};
```

```
NativeDB Introduced: v323
```

## Parameters
* **ped**: The ped handle.
* **ikIndex**: See `eIkPart` for index's.
* **entityLookAt**: Set to 0 for no entity, offsets will be world coordinates instead.
* **boneLookAt**: Set to -1 for no target bone.
* **offsetX**: X-axis offset calculated from the position of `entityLookAt`.
* **offsetY**: Y-axis offset calculated from the position of `entityLookAt`.
* **offsetZ**: Z-axis offset calculated from the position of `entityLookAt`.
* **ikTargetFlags**: See `eIkTargetFlags` for flags.
* **blendInDuration**: Animation blending in (in ms). `-1` by default.
* **blendOutDuration**: Animation blending out (in ms). `-1` by default.

## Examples
```lua
-- Points for 10 seconds
SetIkTarget(PlayerPedId(), 4, 0, 0, -1849.0, -1231.0, 13.0, 0, 0, 10000) 
```
```js
// Points for 10 seconds
SetIkTarget(PlayerPedId(), 4, 0, 0, -1849.0, -1231.0, 13.0, 0, 0, 10000);
```
```cs
using static CitizenFX.Core.Native.API;
// ...

// Points for 10 seconds
SetIkTarget(PlayerPedId(), 4, 0, 0, -1849.0f, -1231.0f, 13.0f, 0, 0, 10000);
```


----

This should fix the order of things wrong, set correctly the descriptions following new method signature, `camelCase` and the use of a prefix like "e" for enums and only one backticks for standart.

This is complete, you can copy/paste.

**Review by PsychoShock — APPROVED** [review]

**Comment by colistro123** [comment]
> Also not sure on the preferred way to do enums is, most use `enum eCamelCase` and its enum variables are `UpperCamelCase` (though this varies from native to native, making it hard to tell what should be done)

Mostly the former, `eCamelCase` that is.

**Review by AvarianKnight — APPROVED** [review]

**Pull request merged**